### DISCUSSION TITLE
Benchmarks

### DISCUSSION THREAD

**Discussion by 0xrnp** [simolus3/drift#3394] [General]
Are there any benchmarks which visualise the strength of drift?

**Comment by simolus3** [comment]
Do you have some particular benchmarks or scenarios in mind?

We have some benchmarks [here](https://github.com/simolus3/drift/tree/develop/extras/benchmarks), but they're a bit stale and are mostly made up of micro-benchmarks that don't necessarily capture a typical drift usage.
Drift definitely adds a bit of overhead to map the lower-level results into the fixed Dart classes for results. Another source of overhead is our use of background isolates (there's some cost of sending messages between isolates), but that's worth it because it reduces load on the main isolate. So drift will be slightly slower than the underlying `sqlite3` package, but we hope that's a price worth paying for the features drift offers.

That said, if you have some particular scenarios in mind that capture how you intend to use drift, I can look into creating an up-to-date benchmark for them.
